src/main/scala/stage/RocketChipAnnotations.scala

```diff
@@ -0,0 +1,84 @@
+// See LICENSE.SiFive for license details.
+
+package freechips.rocketchip.stage
+
+import firrtl.annotations.{Annotation, NoTargetAnnotation}
+import firrtl.options.{HasShellOptions, ShellOption, Unserializable}
+
+sealed trait RocketChipOption extends Unserializable { this: Annotation => }
+
+/* required options */
+
+case class TargetDirectoryAnnotation(targetDir: String) extends NoTargetAnnotation with RocketChipOption
+private[stage] object TargetDirectoryAnnotation extends HasShellOptions {
+  override val options = Seq(
+    new ShellOption[String](
+      longOption = "target",
+      toAnnotationSeq = a => Seq(TargetDirectoryAnnotation(a)),
+      helpText = "<target directory>",
+      shortOption = Some("d")
+    )
+  )
+}
+
+case class TopPackageAnnotation(topModulePackage: String) extends NoTargetAnnotation with RocketChipOption
+private[stage] object TopPackageAnnotation extends HasShellOptions {
+  override val options = Seq(
+    new ShellOption[String](
+      longOption = "top-package",
+      toAnnotationSeq = a => Seq(TopPackageAnnotation(a)),
+      helpText = "<top module package>",
+      shortOption = Some("T")
+    )
+  )
+}
+
+case class TopClassAnnotation(topModuleClass: String) extends NoTargetAnnotation with RocketChipOption
+private[stage] object TopClassAnnotation extends HasShellOptions {
+  override val options = Seq(
+    new ShellOption[String](
+      longOption = "top-class",
+      toAnnotationSeq = a => Seq(TopClassAnnotation(a)),
+      helpText = "<top module class name>",
+      shortOption = Some("t")
+    )
+  )
+}
+
+case class ConfigPackageAnnotation(configsPackage: String) extends NoTargetAnnotation with RocketChipOption
+private[stage] object ConfigPackageAnnotation extends HasShellOptions {
+  override val options = Seq(
+    new ShellOption[String](
+      longOption = "configs-package",
+      toAnnotationSeq = a => Seq(ConfigPackageAnnotation(a)),
+      helpText = "<config classes' package>",
+      shortOption = Some("C")
+    )
+  )
+}
+
+case class ConfigAnnotation(configs: String) extends NoTargetAnnotation with RocketChipOption
+private[stage] object ConfigAnnotation extends HasShellOptions {
+  override val options = Seq(
+    new ShellOption[String](
+      longOption = "configs",
+      toAnnotationSeq = a => Seq(ConfigAnnotation(a)),
+      helpText = "<'_'-delimited config class names>",
+      shortOption = Some("c")
+    )
+  )
```

Comment:
I am keeping `configs` as a single `String`, since this simplifies generating file names later on, but I have moved to using comma-delimiting. The commas are then replaced with underscores to match previous file name generation behavior.

Reply:
If you use the `ShellOption[Seq[String]]` you get native `,` splitting by scopt: https://github.com/scopt/scopt#options. No need to roll your own.

Reply:
I will try that — thanks!

Reply:
> configuration names could include underscores

Looks like for "names" such as `WithJtagDTMSystem_DefaultRV32Config`, these two "_"-separated strings are indeed supposed to be parsed as two separate "config" names. With this in mind, do you think I should still go the route of comma delimiting?

Reply:
After some discussion, I am going to try to stick with comma delimiting and update the Makefiles to adhere to this

Reply:
Cool. I'd be in favor of this, but it obviously depends on your use case and the convention you all want to use.